### ISSUE TITLE
Add option to install apps in nextcloud (using "badges")

### ISSUE THREAD

**Issue by David-Development** [nextcloud/appstore#574]
## Description:

As already discussed in https://github.com/nextcloud/news/pull/318, it would be great to have the option to install apps in your nextcloud right away. Right now when the user wants to install an app he found in the [App Store](https://apps.nextcloud.com/) or on GitHub, the user has to open his/her nextcloud, open the build in app store and search for the app there. 

It would be great for developers to create a so called [Badge](https://play.google.com/intl/en_us/badges/) for their app (e.g. on their GitHub Project page) which will install the app in the users nextcloud installation. This badge can then be integrated into the official nextcloud app store as well. I created a small mockup of the workflow: (Please note that the url for the second step should be something like https://apps.nextcloud.com/...)

As already said in the referenced PR, this will be a huge improvement in the user experience as it will make it much easier for users to install other apps in their nextcloud. Especially considering the recent focus on widening the user base simplifying the installation process makes much sense here. It also allows developers to advertise their apps more/better since it's basically a "one-click" installation. 

Quote by @jancborchardt:
> Then your app store account could also store the address of your Nextcloud, and the install button would then be direct for every install after the first!

## Workflow: 
![](https://user-images.githubusercontent.com/4489723/44958303-ba843780-aede-11e8-9e14-78ee17a31366.png)

**Comment by BernhardPosselt** [comment]
Yeah would work in a similar fashion as the android store but we shouldn't store nextcloud server addresses for privacy and security reasons.

There needs to be a middleman. I'm thinking of something like magnet links, like nextcloud://news which would open your website with the app selected.

**Comment by BernhardPosselt** [comment]
https://developer.mozilla.org/de/docs/Web/API/Navigator/registerProtocolHandler

**Comment by BernhardPosselt** [comment]
Should actually be doable but needs work on the server side

**Comment by David-Development** [comment]
@BernhardPosselt 

> Yeah would work in a similar fashion as the android store but we shouldn't store nextcloud server addresses for privacy and security reasons.

I was thinking that one could use cookies for this? (store last entered url in a cookie?)
Maybe you can even share that cookie with `nextcloud.com` so that when the user is logged in to his nextcloud instance in the background, he'll automatically get the url set for him? (not sure if this is a security issues on the client side though). But I think there is no need to store any information on nextcloud servers (and I agree that we shouldn't do that!).

**Comment by sunjam** [comment]
Happy to see this mentioned.  Home Assistant does this and it is fantastic.  [Example with ESPHome integration](https://www.home-assistant.io/integrations/esphome/).

![image](https://github.com/nextcloud/appstore/assets/1787238/81c5d601-f692-4dd7-adce-99c6c61da6cc)
